src/source-map-tree.ts

```diff
@@ -115,22 +115,32 @@ export default class SourceMapTree {
     if (line >= mappings.length) return null;
 
     const segments = mappings[line];
-    const index = binarySearch(segments, column, segmentComparator);
 
-    // If we can't find an segment that lines up to this column, then we can't
-    // trace it further.
-    if (index < 0) return null;
+    if (segments.length === 0) return null;
+
+    let index = binarySearch(segments, column, segmentComparator);
+
+    // If we can't find an segment that lines up to this column, we use the
+    // closest segment before it.
+    if (index < 0) {
+      index = ~index - 1;
+      if (index < 0) return null; // we come before any mapped segment
+    }
```

Comment:
Let's check this case first:

```suggestion
      if (index === -1) return null; // we come before any mapped segment

    // If we can't find an segment that lines up to this column, we use the
    // closest segment before it.
    if (index < 0) {
      index = ~index - 1;
    }
```